editor/editor_node.cpp

```diff
@@ -1002,7 +1002,12 @@ void EditorNode::save_resource_as(const Ref<Resource> &p_resource, const String
 			continue;
 		}
 		file->add_filter("*." + extensions[i] + " ; " + extensions[i].to_upper());
-		preferred.push_back(extensions[i]);
+
+		if (extensions[i] == "res") {
+			preferred.push_back(extensions[i]);
+		} else {
+			preferred.push_front(extensions[i]);
+		}
```

Comment:
Does this still give `*.tres` priority as the default, text-based extension, or is it dependent on inserting order?

Reply:
I guess it depends on the inserting order. Maybe it is a good idea to ensure .tres is actually preferred just in case the insertion order changes of other formats are introduced in the future.